### DISCUSSION TITLE
getStaticPaths with large amount of items

### DISCUSSION THREAD

**Discussion by jcamato** [vercel/next.js#27574] [Help]
I recently started trying to switch my CRA to Next, and I had one question pertaining to dynamic routes and getStaticPaths. If my database has over 600,000 movies, do I still have to pass an array of every single movie ID for the individual movie routes (movie/[id])? This seems like it will unnecessarily decrease performance with an additional server call when I don't actually need to statistically generate every single route in the background.

**Comment by raskardev** [comment]
You can use the [`fallback`](https://nextjs.org/docs/basic-features/data-fetching#the-fallback-key-required) prop and set it to `true` so you can pass, for example, 100 top movie IDs to getStaticPaths. Next will build remaining pages in the background on demand and then swaps from fallback page to the compiled page.
